apps/passport-server/src/services/poapService.ts

```diff
@@ -205,13 +287,54 @@ export class PoapService {
     }
   }
 
+  /**
+   * Given a ZKEdDSAEventTicketPCD sent to the server for claiming a Zuzalu 2023 POAP,
+   * returns the valid redirect URL to the response handler.
+   *  1. If this ticket is already associated with a POAP mint link, return that link.
+   *  2. If this ticket is not associated with a POAP mint link and more unclaimed POAP
+   *     links exist, then associate that unclaimed link with this ticket and return it.
+   *  3. If this ticket is not associated with a POAP mint link and no more unclaimed
+   *     POAP links exist, return a custom server error URL.
+   */
+  public async getZuzalu23PoapRedirectUrl(
+    serializedPCD: string
+  ): Promise<string> {
+    try {
+      const ticketId = await this.validateZuzalu23Ticket(serializedPCD);
+      const poapLink = await this.getPoapClaimUrlByTicketId(
+        ticketId,
+        "zuzalu23"
+      );
+      if (poapLink == null) {
+        throw new Error("Not enough Zuzalu 2023 POAP links");
+      }
+      return poapLink;
+    } catch (e) {
+      logger("[POAP] getZuzalu23PoapRedirectUrl error", e);
+      this.rollbarService?.reportError(e);
+      // Return the generic /server-error page instead for the route to redirect to,
+      // with a title and description informing the user to contact support.
+      return getServerErrorUrl(
+        "Contact support",
+        "An error occurred while fetching your POAP mint link for Zuzalu 2023."
+      );
```

Comment:
does this error page explain the best way to contact support?

Reply:
Yes, see https://zupass.org/#/server-error?title=Contact+support&description=An+error+occurred+while+fetching+your+POAP+mint+link+for+Zuzalu+2023.

<img width="373" alt="Screenshot 2024-01-17 at 4 36 40 PM" src="https://github.com/proofcarryingdata/zupass/assets/36896271/5639c3d9-e9d0-4027-bd81-44c37c76d849">